src/main/java/org/apache/maven/plugins/jlink/JLinkMojo.java

```diff
@@ -428,11 +437,13 @@ private JLinkExecutor getExecutor()
         return getJlinkExecutor();
     }
 
-    private boolean projectHasAlreadySetAnArtifact()
+    private boolean projectHasAlreadySetAnArtifact( File createZipArchiveFromImage )
     {
-        if ( getProject().getArtifact().getFile() != null )
+        Optional<File> file = Optional.ofNullable( getProject().getArtifact().getFile() );
```

Comment:
optional here reads even more opaquely than simply checking for null

Reply:
Actually we're missing an `isFile` check here. Thanks for catching this